core/src/main/kotlin/org/evomaster/core/search/gene/Gene.kt

```diff
@@ -69,12 +72,29 @@ abstract class Gene(var name: String) {
      *   @param allGenes if the gene depends on the other (eg a Foreign Key in SQL databases),
      *          we need to refer to them
      *   @param evi the evaluated individual contains an evolution of the gene with fitness values
+     *   @param selection how to select genes to mutate if [this] contains more than one genes(e.g., ObjectGene) or other characteristics(e.g., size of ArrayGene)
+     *   @param impact info of impact of the gene if it has, but in some case impact might be null, e.g., an element at ArrayGene
+     *   @param geneReference a reference (i.e., id generated) to find a gene in this history, which always refers to 'root' gene in the [evi]
+     *   @param archiveMutator mutate genes using archive-based methods if the method is enabled or supports this type of [this] gene.
      */
     open fun archiveMutation(randomness: Randomness,
                              allGenes: List<Gene>,
                              apc: AdaptiveParameterControl,
-                             evi: EvaluatedIndividual<*>){
-        TODO("not implemented")
+                             selection: GeneMutationSelectionMethod,
+                             impact: GeneImpact?,
+                             geneReference: String,
+                             archiveMutator: ArchiveMutator,
+                             evi: EvaluatedIndividual<*>,
+                             targets: Set<Int>){
+        standardMutation(randomness, apc, allGenes)
+//        if (!archiveMutator.enableArchiveMutation()){
+//            standardMutation(randomness,apc, allGenes)
+//        }else{
+//            if (archiveMutator.doesSupport(this))
+//                archiveMutator.mutate(this)
+//            else
+//                TODO("ArchiveGeneMutation for ${this::class.java.simpleName} was NOT IMPLEMENT")
+//        }
```

Comment:
i guess this commented code could be removed?
plus could add the comment: "if this method is not overridden, just default to standard mutation"